content/posts/2021/01/2021-11-04-Getting-Started-With-React/index.md

```diff
@@ -0,0 +1,334 @@
+
+This is where you get the line ``installing react, react-dom and react scripts``. The first thing it does is check if there's ``package.json`` file and ``yarn.lock``. It loads ``package.json`` file into a variable called ``appPackage`` in JSON format. It checks if TypeScript is a dependency after it does this. Next it begins adding to the JSON.
+
+First is the scripts, this will probably be familiar as ``start``, ``build``, ``eject`` and ``test``. These are located in init.js file that we are currently going through. Next is the eslintConfig and defaultBrowsers. These are by default `>0.2%','not dead', 'not ie <= 11','not op_mini all'. Basically this means target all browsers that are higher than .2% market share, in active use, not below IE 11 or Opera Mini.This is a seperate package located in ``package.json``. You can find it  [<span> Here </span>](https://www.npmjs.com/package/browserslist).
+
+The ``package.json`` file is written out and will not be changed again. At this point there's no output yet, a ``README.md`` file is checked and if it exists it is renamed. The files specified in the template folder are then copied to a working directory. This is the ``public``, ``README.me``, ``src`` and ``gitignore``. ``gitignore`` is copied without the ``.`` infront then renamed, this is to prevent an npm bug.
+
+After a check to see if React is installed, it moves to check if the repository has been initialized with git. These runs git init, and adds the first commit. This is where the line ``Initialized a git repository`` comes from.
+
+The last thing init.js does before ending the process is print the success message and instructions. These introduce you to basic ``react scripts`` commands and suggest you cd into your app directory to run your application.
+
+
+### Index.html
+
+This is the entry file, the first thing that the web browser loads when a user navigates to your app. If you look at the file, it's just a normal HTML file that you're hopefully familiar with. If you look at the body, it's empty. React dynamically convert our React code into HTML and load it here, in the div with id "root". With that out of the way let's get to the juicy part.
+
+
+## Our First Component
+Open up our **App.js** component, this is the main component in our application. It is the first component to be rendered, the icing to our cake.
+
+The first thing we are going to do with our icing is remove all the code so that we build our component from scratch so that we better understand what's going on. 
+
+Now that we have a nice and clean slate,we will start by importing ``react``
+
+``import react from 'react';``
+
+Next we declare a function, we will use ES6 arrow function here. That's more or less what a component is...A function with some logic and markup ( In this case JSX ). We are also going to export this function so we can use it elsewhere.
+
+        `` const App = () =>{
```

Comment:
Improper code block, use 3 back-ticks followed by language -`js` in this case -, then new line, then the code snippet then new line then 3 more back-ticks

Example:

![Screenshot 2021-11-28 at 23 22 09](https://user-images.githubusercontent.com/9129843/143784541-bc7c47d7-a412-4e37-b9e4-560e21f37ac8.png)

Reply:
Closing this..as They were all fixed..